dts/bindings/gpio/nxp,pcal6416a.yaml

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2023 SILA Embedded Solutions GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+description: PCAL6416A 16-bit I2C-based I/O expander
+
+compatible: "nxp,pcal6416a"
+
+include: [i2c-device.yaml, gpio-controller.yaml]
+
+properties:
+  ngpios:
+    required: true
+    const: 16
```

Comment:
I would suggest making this non-const so that individual device tree files can specify how many gpios are available for a given system.

Reply:
Please also add something like a `chip_type` property and a header somewhere that defines something like this (it can be included via device tree)
```
#define PCAL6408A_CHIP_TYPE 0
#define PCAL6416A_CHIP_TYPE 1
```

Reply:
I disagree. Using two different bindings with `const` is okay. Please see the discussion here: https://github.com/zephyrproject-rtos/zephyr/pull/54433#discussion_r1118422176

I would suggest introducing a `nxp,pcal64xxa-base.yaml` file with the common properties.